Deployment/asdk-prechecker.ps1

```diff
@@ -0,0 +1,384 @@
+        $Global:ChecksFailure++
+    }
+    else
+    {
+        write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- The host OS version matches the requirements for ASDK ($server2016OSVersionRequired)."
+        $Global:ChecksSuccess++
+    }
+}
+
+function CheckDomainJoinStatus {
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking domain join status..."
+
+    $sysInfo = Get-WmiObject -Class Win32_ComputerSystem
+    if ($sysInfo.PartOfDomain) {
+        write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- The host must not be domain joined. Please leave the domain and try again."
+        $Global:ChecksFailure++
+    }
+    else
+    {
+        write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- The host is not domain joined."
+        $Global:ChecksSuccess++
+    }
+
+}
+
+function CheckVMSwitch {
+
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking NIC status..."
+
+    if (([array](Get-NetAdapter | ? {$_.Status -eq 'Up'})).Count -ne 1) {
+        write-host -ForegroundColor darkyellow "["(date -format "HH:mm:ss")"]" " -- Multiple NICs, virtual switches or NIC teaming are not allowed. Please only keep one physical NIC enabled and remove virtual switches or NIC teaming. This message can be ignored if you are planning to leverage the ASDK Installer from GitHub, as it provides a way to configure the NICs."
+        $Global:ChecksSuccess++
+    }
+    else
+    {
+        write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- NIC configuration passed successfully."
+        $Global:ChecksSuccess++
+    }
+}
+
+function CheckServerName {
+
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking server name..."
+
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Server name is" $Env:COMPUTERNAME
+
+  if ($Env:COMPUTERNAME -eq 'AzureStack') {
+    write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Server name cannot be ""AzureStack"" since it conflicts with the domain name."
+    $Global:ChecksFailure++
+  }
+  else
+  {
+    write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- Server name does not conflict with future domain name AzureStack.local."
+    $Global:ChecksSuccess++
+  }
+}
+
+
+function CheckCPU {
+
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking processor information..."
+
+    $CPUCount = (Get-WmiObject -class win32_processor –computername localhost).count
+    $CoreCount =  ((Get-WmiObject -class win32_processor –computername localhost -Property "numberOfCores")[0].numberOfCores)*$CPUCount
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Number of CPU sockets = $CPUCount"
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Number of physical cores =  $CoreCount"
+
+    If (($CPUCount -lt 2) -or ($CoreCount -lt 12)){
```

Comment:
Currently the CPU requirement for ASDK is with "16" minimum and "20" recommended. Please update accordingly

Reply:
Changed test condition to `$CoreCount -lt 16`. Updated error message description.